starfish/imagestack/imagestack.py

```diff
@@ -290,6 +290,131 @@ def xarray(self) -> xr.DataArray:
         """Retrieves the image data as an xarray.DataArray"""
         return self._data
 
+    def __getitem__(self, key) -> "ImageStack":
+        # index xarray (keep dims) and create new stack
+        indexed_data = self._index_keep_dimensions(self.xarray, key)
+        stack = self.from_numpy_array(indexed_data.data)
+        # set coords on new stack
+        stack._coordinates = self._calc_new_coords(indexers=self._item_key_to_dict(key))
+        return stack
+
+    def _item_key_to_dict(self, key):
```

Comment:
I actually think it would be easier if you take each index value, and expanded it to the actual values, i.e., `slice(None)` -> `slice(0, sz-1)`.  your code for recalculating the coordinates would probably be isolated from the weirdness of slice.